src/test/java/com/epam/service/impl/TraineeServiceImplTest.java

```diff
@@ -1,122 +1,96 @@
-        Assertions.assertEquals(createdTrainee.getUser().getFirstName(), trainee.getUser().getFirstName());
-        Assertions.assertEquals(createdTrainee.getUser().getLastName(), trainee.getUser().getLastName());
-        Assertions.assertEquals(createdTrainee.getUser().getPassword(), trainee.getUser().getPassword());
-        Assertions.assertEquals(createdTrainee.getUser().isActive(), trainee.getUser().isActive());
-    }
+    void testFindByUsername() {
+        TraineeJsonDto traineeJsonDto = TraineeMockData.getMockedTrainee_2().get(0);
+        Trainee mockedTrainee = Mappers.convertTraineeJsonDtoToTrainee(traineeJsonDto);
+        when(traineeDAO.findByUsername(mockedTrainee.getUser().getUsername())).thenReturn(mockedTrainee);
+
+        Trainee result = traineeService.getTraineeByUsername(mockedTrainee.getUser().getUsername());
+
+        Assertions.assertNotNull(result);
+        Assertions.assertEquals(mockedTrainee, result);
+
+
+        Assertions.assertEquals(result.getId(), mockedTrainee.getId());
+        Assertions.assertEquals(result.getId(), traineeJsonDto.getId());
+        Assertions.assertEquals(result.getAddress(), mockedTrainee.getAddress());
+        Assertions.assertEquals(result.getDateOfBirth(), mockedTrainee.getDateOfBirth());
+        Assertions.assertEquals(result.getTrainings(), mockedTrainee.getTrainings());
+        Assertions.assertEquals(result.getTrainers(), mockedTrainee.getTrainers());
+        Assertions.assertEquals(result.getUser().getId(), mockedTrainee.getUser().getId());
+        Assertions.assertEquals(result.getUser().getId(), traineeJsonDto.getUser().getId());
+        Assertions.assertEquals(result.getUser().getFirstName(), mockedTrainee.getUser().getFirstName());
+        Assertions.assertEquals(result.getUser().getLastName(), mockedTrainee.getUser().getLastName());
+        Assertions.assertEquals(result.getUser().getPassword(), mockedTrainee.getUser().getPassword());
+        Assertions.assertEquals(result.getUser().isActive(), mockedTrainee.getUser().isActive());
 
-    @Test
-    public void testDeleteTrainee() {
-        Trainee trainee = new Trainee();
-        trainee.setId(1);
-        trainee.setAddress("Address");
-        traineeService.createTrainee(trainee);
-        traineeService.deleteTrainee(1);
-
-        Assertions.assertNull(traineeService.getTraineeById(1));
     }
 
     @Test
-    public void testUpdateTrainee() {
-        Trainee trainee = new Trainee();
-        trainee.setAddress("Address");
-        traineeService.createTrainee(trainee);
-        Trainee createdTrainee = traineeService.getAllTrainees().get(0);
-        createdTrainee.setAddress("Updated Address");
-        traineeService.updateTrainee(createdTrainee);
-
-        Assertions.assertEquals(createdTrainee.getAddress(), trainee.getAddress());
+    void testCheckUsernameAndPasswordMatching() {
+        // Given
+        String username = "testUser";
+        String password = "testPassword";
```

Comment:
You miss a lot of things here. Please add tests for cases when 
```
String username = null;
String password = null;
```
and all of combinations, like 1) username is null, but password is not null, 2) username is not null, but password is null etc.